host-interaction/file-system/reference-absolute-stream-path-on-windows.yml

```diff
@@ -4,19 +4,23 @@ rule:
     namespace: host-interaction/file-system
     authors:
       - blas.kojusner@mandiant.com
+      - william.ballenthin@mandiant.com
     scope: basic block
     references:
       - https://learn.microsoft.com/en-us/windows/win32/fileio/file-streams
     examples:
       - 51828683DC26BFABD3994494099AE97D:0x11A9
   features:
     - and:
-      - string: /(..\?\\)?([\w]\:|\\)(\\((?![\<\>\"\/\|\*\?])[\x20-\x7E])+)+:(((?![\<\>\"\/\|\*\?])[\x20-\x7E])+)+$/
+      - string: /(..\?\\)?([\w]\:|\\)(\\+((?![\<\>\"\/\|\*\?\\])[\x20-\x5B\x5D-\x7E])+)+:(((?![\<\>\"\/\|\*\?\\])[\x20-\x5B\x5D-\x7E])+)+$/
         # (..\?\\)? -> Check for path starting with "\\?\"
         # ([\w]\:|\\) -> Check for absolute path beginning
-        # (\\((?![\<\>\"\/\|\*\?])[\x20-\x7E])+)+ -> Check for valid path and filename
+        # (\\+((?![\<\>\"\/\|\*\?\\])[\x20-\x5B\x5D-\x7E])+)+ -> Check for valid path and filename
+        #   \\+ -> Check for at least one backslash path separator
+        #   (?![\<\>\"\/\|\*\?\\]) -> path component must not start with <, >, ", ...
+        #   [\x20-\x5B\x5D-\x7E] -> path component must be printable ASCII, except backslash path separator
         # : -> Check for start of stream filename
-        # (((?![\<\>\"\/\|\*\?])[\x20-\x7E])+)+ -> Check for valid stream filename
+        # (((?![\<\>\"\/\|\*\?\\])[\x20-\x5B\x5D-\x7E])+)+ -> Check for valid stream filename
```

Comment:
```suggestion
        # \$?([a-zA-Z0-9_]+))?$ -> Check for valid stream filename
```

Reply:
do you have a reference for the valid characters for stream names? e.g. i dont see space allowed here and i dont know if thats a mistake or intentional.

Reply:
hm, per MSDN: "Any characters that are legal for a file name are also legal for the stream name, including spaces. For more information, see [Naming a File](https://learn.microsoft.com/en-us/windows/win32/fileio/naming-a-file)." so yeah, maybe better to stick to the prior definition.

Reply:
```ps1
PS C:\Users\william.ballenthin> set-content -stream "space bar" -value "bar" .\a.txt
PS C:\Users\william.ballenthin> get-item .\a.txt -stream *


PSPath        : Microsoft.PowerShell.Core\FileSystem::C:\Users\william.ballenthin\a.txt::$DATA
PSParentPath  : Microsoft.PowerShell.Core\FileSystem::C:\Users\william.ballenthin
PSChildName   : a.txt::$DATA
PSDrive       : C
PSProvider    : Microsoft.PowerShell.Core\FileSystem
PSIsContainer : False
FileName      : C:\Users\william.ballenthin\a.txt
Stream        : :$DATA
Length        : 12

PSPath        : Microsoft.PowerShell.Core\FileSystem::C:\Users\william.ballenthin\a.txt:bar
PSParentPath  : Microsoft.PowerShell.Core\FileSystem::C:\Users\william.ballenthin
PSChildName   : a.txt:bar
PSDrive       : C
PSProvider    : Microsoft.PowerShell.Core\FileSystem
PSIsContainer : False
FileName      : C:\Users\william.ballenthin\a.txt
Stream        : bar
Length        : 5

PSPath        : Microsoft.PowerShell.Core\FileSystem::C:\Users\william.ballenthin\a.txt:space bar
PSParentPath  : Microsoft.PowerShell.Core\FileSystem::C:\Users\william.ballenthin
PSChildName   : a.txt:space bar
PSDrive       : C
PSProvider    : Microsoft.PowerShell.Core\FileSystem
PSIsContainer : False
FileName      : C:\Users\william.ballenthin\a.txt
Stream        : space bar
Length        : 5
```

Reply:
`$` seems to be ok, `:` and slashes are not. at least from powershell.